std/range/package.d

```diff
@@ -10899,7 +10899,20 @@ cost $(BIGOH n), use $(REF isSorted, std,algorithm,sorting).
 auto assumeSorted(alias pred = "a < b", R)(R r)
 if (isInputRange!(Unqual!R))
 {
-    return SortedRange!(Unqual!R, pred)(r);
+    // Avoid senseless `SortedRange!(SortedRange!(...), pred)` nesting.
+    static if (is(R == SortedRange!(RRange, RPred), RRange, alias RPred))
+    {
+        static if (isInputRange!R && __traits(isSame, pred, RPred))
+            // If the predicate is the same and we don't need to cast away
+            // constness for the result to be an input range.
+            return r;
+        else
+            return SortedRange!(Unqual!(typeof(r._input)), pred)(r._input);
```

Comment:
This is never covered. Sth. like this should cover it:

```
[0, 1, 2].assumeSorted!((a, b) => a == b).assumeSorted!((a, b) => a + 1 == b +1);
```

Reply:
Fixed.